tests/operators/big_int.cpp

```diff
@@ -0,0 +1,519 @@
+          for(i=l1-1;i>=0;i--) {
+         carry = 0;
+         int n=str1[i]-'0';
+        for (j=l2-1;j>=0;j--) {
+            int s = number[i+j+1] + carry + (n)*(str2[j]-'0');
+            //cout<<s<<endl;
+            number[i+j+1] = s%10;
+            carry = s/10;
+               //cout<<s<<endl;
+          }
+          first=i+j+2;
+          if(carry)
+          {
+            number[i+j+1]+=carry;
+            first--;
+          }
+    }  
+          string s = ""; 
+      int i1=first;
+    while (i1<(l1+l2)) 
+       { 
+        s +=(number[i1++]+'0'); 
+       } ans.str=s;
+           return ans;
+}
+else
+{BigInteger res("0");
+  string s[10];
+s[0]="0";
+s[1]=str1;
+
+int l=str1.size();
+for(int i=2;i<=9;i++)
+{ string result;
+  int mul=0;
+for (int j=l-1;j>=0;j--) {
+            mul += i*(str1[j]-'0');
+            //cout<<s<<endl;
+             result.insert(0,1,((mul%10)+'0'));
+              mul=mul/10;
+               //cout<<s<<endl;
+          }
+          if(mul)
+            result.insert(0,1,(mul+'0'));
+          s[i]=result;
+          //cout<<s[i]<<endl;
+}
+int j=0;
+
+for(int i=l2-1;i>=0;i--)
+{
+  string temp=s[str2[i]-'0'];
+  //cout<<temp<<" ";
+  temp.insert(temp.end(),l2-1-i,'0');
+  //cout<<res<<" ";
+  //BigInteger temp1(temp);
+  res=res+temp;
+  //cout<<temp1<<" "<<res<<endl;
+j++;
+}
+return res;}
+
+            
+              }*/
+
+
+/*BigInteger operator* (BigInteger  a, BigInteger  b) {
+
+BigInteger res("0");
+string str1=a.str;
+string str2=b.str;
+string s[10];
+s[0]="0";
+s[1]=str1;
+
+int l=str1.size()-1;
+for(int i=2;i<=9;i++)
+{ string result;
+  int mul=0;
+for (int j=l;j>=0;j--) {
+            mul += i*(str1[j]-'0');
+            //cout<<s<<endl;
+             result.insert(0,1,((mul%10)+'0'));
+              mul=mul/10;
+               //cout<<s<<endl;
+          }
+          if(mul)
+            result.insert(0,1,(mul+'0'));
+          s[i]=result;
+          //cout<<s[i]<<endl;
+}
+int j=0;
+for(int i=str2.size()-1;i>=0;i--)
+{
+  string temp=s[str2[i]-'0'];
+  //cout<<temp<<" ";
+  temp=temp.append(j,'0');
+  //cout<<res<<" ";
+  //BigInteger temp1(temp);
+  res=res+temp;
+  //cout<<temp1<<" "<<res<<endl;
+j++;
+}
+return res;
+}*/
+
+/*
+BigInteger operator/ (BigInteger  a, BigInteger  b) {
+        //string s="";
+        BigInteger div("0");
+        if(a.sign==1 && b.sign==1){ }
+       else if(a.sign ==1){ div.sign=1;}
+      else if(b.sign ==1){ div.sign=1;}
+     //if(b>a){return b/a;} 
+       
+       //BigInteger c=a;
+       // cout<<c<<" "<<b<<" "<<endl;
+       while( a >= b )
+       {
+        //bool bl = c>=b ;
+          a=a-b;
+          div=div+1;
+          cout<<a<<" "<<b<<endl;
+        }
+
+           return div;
+            
+              }
+*/
+
+BigInteger operator% (BigInteger  a, BigInteger  b) {
+        //string s="";
+
+        BigInteger mod("0");
+      string str1=b.str;
+      int si=str1.size();
+      BigInteger temp("0");
```

Comment:
Codacy found an issue: [Variable 'temp' is assigned a value that is never used.](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6290647)